### ISSUE TITLE
Problems running on Fedora 23

### ISSUE THREAD

**Issue by mynamewastaken** [pantheon-cookbooks/dnf#16]
I suspect this may be related to https://github.com/pantheon-cookbooks/dnf/issues/14 , but to get this to run successfully I had 2 separate issues:
- The `dnf-query.py` script seems to be missing when pulling the cookbook down via Berkshelf or knife cookbook site install etc.

```
[2016-02-26T17:51:57-06:00] INFO: Processing dnf_package[git] action install ($COOKBOOK::laptop line 37)

================================================================================
Error executing action `install` on resource 'dnf_package[git]'
================================================================================

Errno::ENOENT
-------------
No such file or directory - $PATH/git/chef/vendor/dnf/libraries/dnf-query.py

Cookbook Trace:
---------------
$PATH/git/chef/vendor/dnf/libraries/provider_dnf_package.rb:112:in `available_version'
$PATH/git/chef/vendor/dnf/libraries/provider_dnf_package.rb:77:in `block in load_current_resource'
$PATH/git/chef/vendor/dnf/libraries/provider_dnf_package.rb:75:in `each'
$PATH/git/chef/vendor/dnf/libraries/provider_dnf_package.rb:75:in `load_current_resource'

Resource Declaration:
---------------------
# In $PATH/git/chef/vendor/$COOKBOOK/recipes/laptop.rb

 37:   package pkg do
 38:     action :install
 39:   end
 40: end

Compiled Resource:
------------------
# Declared in $PATH/git/chef/vendor/$COOKBOOK/recipes/laptop.rb:37:in `block in from_file'

dnf_package("git") do
  package_name "git"
  provider Chef::Provider::Package::Dnf
  action [:install]
  retries 0
  retry_delay 2
  default_guard_interpreter :default
  declared_type :package
  cookbook_name :$COOKBOOK
  recipe_name "laptop"
end

[2016-02-26T17:51:57-06:00] INFO: Running queued delayed notifications before re-raising exception
[2016-02-26T17:51:57-06:00] ERROR: Running exception handlers
[2016-02-26T17:51:57-06:00] ERROR: Exception handlers complete
[2016-02-26T17:51:57-06:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2016-02-26T17:51:57-06:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2016-02-26T17:51:57-06:00] ERROR: dnf_package[git] ($COOKBOOK::laptop line 37) had an error: Errno::ENOENT: No such file or directory - $PATH/git/chef/vendor/dnf/libraries/dnf-query.py
[2016-02-26T17:51:57-06:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```
- Have to `dnf install python-dnf`, else you get the below error:

```
[2016-02-26T17:55:58-06:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2016-02-26T17:55:58-06:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2016-02-26T17:55:58-06:00] ERROR: dnf_package[git] ($COOKBOOK::laptop line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of $PATH/git/chef/vendor/dnf/libraries/dnf-query.py git ----
STDOUT: 
STDERR: Traceback (most recent call last):
  File "$PATH/git/chef/vendor/dnf/libraries/dnf-query.py", line 48, in <module>
    import dnf
ImportError: No module named dnf
---- End output of $PATH/git/chef/vendor/dnf/libraries/dnf-query.py git ----
Ran $PATH/chef/vendor/dnf/libraries/dnf-query.py git returned 1
```

On a side note, maybe a quick note on how to use this cookbook would be nice. I had to do some research.

**Comment by sjsadowski** [comment]
Also confirmed on F22.

**Comment by spheromak** [comment]
This is a known Issue because chef server filters the dnf-query.py out.  We haven't come up with a good solution for shipping this in a chef-server setup. We run chef-solo internally, and it doesn't effect us. Long term the hope was that this would get adopted into chef core, but we haven't been pushing that effort.

**Comment by knightsamar** [comment]
Reporting this issue on Fedora 24 as well.

**Comment by lamont-granquist** [comment]
should have been fixed in core in 12.8.2:  https://github.com/chef/chef/commit/b62e6ae14ddd9c60035f4ec314f4c6d0ab47777e

**Comment by VanAxe** [comment]
I'm having the same issue on Fedora 24 as well.

```plaintext
$ uname -a
Linux default-fedora-24 4.5.5-300.fc24.x86_64 #1 SMP Thu May 19 13:05:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

$ chef-client --version
Chef: 12.16.42
```

```plaintext
           ================================================================================
           Error executing action `install` on resource 'dnf_package[/tmp/kitchen/cache/rpmfusion-free-release-24.noarch.rpm]'
           ================================================================================
           
           Errno::ENOENT
           -------------
           No such file or directory - /tmp/kitchen/cache/cookbooks/dnf/libraries/dnf-query.py
           
           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/dnf/libraries/provider_dnf_package.rb:112:in `available_version'
           /tmp/kitchen/cache/cookbooks/dnf/libraries/provider_dnf_package.rb:77:in `block in load_current_resource'
           /tmp/kitchen/cache/cookbooks/dnf/libraries/provider_dnf_package.rb:75:in `each'
           /tmp/kitchen/cache/cookbooks/dnf/libraries/provider_dnf_package.rb:75:in `load_current_resource'
```
